typescript/infra/src/xerc20/utils.ts

```
@@ -279,24 +286,87 @@ async function prepareRateLimitTx(
+  await getSafeTxService(chain, multiProvider, bridgeAddress);
+
   try {
-    return await canProposeSafeTransactions(
-      proposer,
-      chain,
-      multiProvider,
-      safeAddress,
+    await getSafe(chain, multiProvider, safeAddress);
+    rootLogger.info(
+      chalk.gray(`[${chain}][${bridgeAddress}] Safe found: ${safeAddress}`),
     );
+    return true;
   } catch {
+    rootLogger.info(
+      chalk.gray(`[${chain}][${bridgeAddress}] Safe not found: ${safeAddress}`),
+    );
+    return false;
+  }
+}
+
+async function checkSafeProposer(
+  proposer: Address,
+  chain: string,
+  multiProvider: MultiProvider,
+  safeAddress: Address,
+  bridgeAddress: Address,
+): Promise<boolean> {
+  const safeService = await getSafeTxService(
+    chain,
+    multiProvider,
+    bridgeAddress,
+  );
+  // TODO: assumes the safeAddress is in fact a safe
+  const safe = await getSafe(chain, multiProvider, safeAddress);
+
+  const delegates = await getSafeDelegates(safeService, safeAddress);
+  const owners = await safe.getOwners();
+
+  const isSafeProposer =
+    delegates.includes(proposer) || owners.includes(proposer);
+
+  if (isSafeProposer) {
+    rootLogger.info(
+      chalk.gray(
+        `[${chain}][${bridgeAddress}] Safe proposer detected: ${proposer}`,
+      ),
+    );
+    return true;
+  } else {
+    rootLogger.info(
+      chalk.gray(
+        `[${chain}][${bridgeAddress}] Safe proposer not detected: ${proposer}`,
+      ),
+    );
     return false;
   }
 }
 
+async function getSafeTxService(
+  chain: string,
+  multiProvider: MultiProvider,
+  bridgeAddress: Address,
+): Promise<any> {
+  let safeService;
+  try {
+    safeService = getSafeService(chain, multiProvider);
+  } catch (error) {
+    rootLogger.error(
+      chalk.red(
+        `[${chain}][${bridgeAddress}] Safe service not available, cannot send safe transactions, please add the safe service url to registry and try again.`,
+      ),
+    );
+    throw { chain, error };
```

throw if safe tx service not set and avoid silently falling back to Signer txs @nambrot